code/modules/power/solar.dm

```diff
@@ -51,7 +52,24 @@
 	else
 		S.forceMove(src)
 	if(S.glass_type == /obj/item/stack/sheet/rglass) //if the panel is in reinforced glass
-		max_integrity *= 2 								 //this need to be placed here, because panels already on the map don't have an assembly linked to
+		max_integrity *= 1.2								 //this need to be placed here, because panels already on the map don't have an assembly linked to
+		glass_effect = 1.2
+		obj_integrity = max_integrity
+	if(S.glass_type == /obj/item/stack/sheet/plasmaglass) //if the panel is in plasma glass
```

Comment:
At this point I think you should make a new sheet variable (hereby called `solars_multiplier`, default value 1) and replace all this if cascade with:
```
	var/pmultiplier = initial(S.glass_type.solars_multiplier)
	max_integrity *= pmultiplier 
	power_multiplier = pmultiplier 
```
While also specifying such variable values for the other special glass sheets, of course.

By the by, to make this work you'd also have to change this line:
https://github.com/Citadel-Station-13/Citadel-Station-13/blob/d75a9cf36815aed3ed5519f1a1b3b7383213fa46/code/modules/power/solar.dm#L189
to 
```
	var/obj/item/sheet/glass_type
```
Else the `initial()` call won't work.

Reply:
Dosnt work